### ISSUE TITLE
Configuration of default HTTP server broken in version 5.2.1

### ISSUE THREAD

**Issue by janskarvall** [sous-chefs/apache2#568]
### Cookbook version
5.2.1

### Chef-client version
14.2.0

### Platform Details
ubuntu-16.04.5-server-amd64

### Scenario:
Configure apache default HTTP site.

### Steps to Reproduce:
Include the following lines in a recipe, put it in the run-list and cook:

```
include_recipe 'apache2'
apache_site '000-default' do
    enable true
end
```
 

### Expected Result:
Accessing the server with http://_servername_/ with a browser shows the `Apache2 Ubuntu Default Page`.

### Actual Result:
The browser shows the following message:

```
Forbidden

You don't have permission to access / on this server.
```
### Some findings
apache2 version 5.0.1 cookbook works fine, and the `Apache2 Ubuntu Default Page` is shown.

After having cooked with version 5.2.1 and then, when cooking with version 5.0.1, the following changes in the `/etc/apache2/apache2.conf` file are shown:

```
  * template[apache2.conf] action create
    - update content in file /etc/apache2/apache2.conf from dff6d5 to d11711
    --- /etc/apache2/apache2.conf	2018-11-11 15:56:47.970025852 +0100
    +++ /etc/apache2/.chef-apache220181111-17263-1hqmeff.conf	2018-11-11 16:03:08.544667883 +0100
    @@ -40,8 +40,10 @@
     #
     KeepAliveTimeout 5
     
    +#<IfModule unixd_module>
     User www-data
     Group www-data
    +#</IfModule>
     
     # Sets the default security model of the Apache2 HTTPD server. It does
     # not allow access to the root filesystem outside of /usr/share and /var/www/html.
    @@ -60,11 +62,18 @@
             Require all denied
     </Directory>
     
    +<Directory /usr/share>
    +        AllowOverride None
    +        Require all granted
    +</Directory>
    +
    +<Directory /var/www/html>
    +        Options Indexes FollowSymLinks
    +        AllowOverride None
    +        Require all granted
    +</Directory>
    +
     #
    -# No DocumentRoot or other <Directory /> blocks are declared in this
    -# file. These statements should be covered in virtual site files created
    -# in "#{node['apache']['dir']}/sites-available.
    -#
     # AccessFileName: The name of the file to look for in each directory
     # for additional configuration directives.  See also the AllowOverride
     # directive.
    @@ -115,7 +124,6 @@
     
     # Include ports listing
     Include /etc/apache2/ports.conf
    -
     
     #
     # The following directives define some format nicknames for use with
```
There are also these lines in the output of the cooking with version 5.2.1:

```
Recipe: apache2-cookbook-5.2.1-default-site-bug::default
  * execute[a2ensite 000-default.conf] action run (skipped due to only_if)
```

Also, I find, that after cooking with version 5.2.1, there is no `000-default.conf` file generated:

```
# ls /etc/apache2/sites-available/
default-ssl.conf
```

I get the impression that generation of `000-default.conf` file has for some reason failed and that the the lines "missing" in the `apache.conf` file, should now be included in that file. 

This is all guessing, as the apache2 `README.md` file does not clearly describe how available sites are handled, which names they have, and how to use them.

**Comment by damacus** [comment]
Hey there! 

We're looking at moving to a fully custom resource driven model for the apache2 cookbook. 
As a result our ability to backport fixes into the current release safely is limited.

**Comment by damacus** [comment]
Closing this as a `won't fix`, as I'm focusing on the 6.0 series of bugs I'm about to create.
